ibis/expr/schema.py

```diff
@@ -67,6 +67,10 @@ def types(self):
     def geospatial(self) -> tuple[str, ...]:
         return tuple(name for name, typ in self.fields.items() if typ.is_geospatial())
 
+    @attribute
+    def null_fields(self) -> tuple[str, ...]:
```

Comment:
This should be done in a follow-up.

Reply:
In the followup, the semantics of this are going to need to change. Since this is internal-only, I think it is fine for this churn to happen, but just writing this down for when I/we implement it.

We use this for two things:

1. To check whether we need to do the pyarrow fixup for duckdb, eg if any field or sub-field is null.
2. To provide nice error messages when trying to create_table() a table-with-null-types, eg 

```python
        if null_columns := schema.null_fields:
            raise com.IbisTypeError(
                f"{self.name} cannot yet reliably handle `null` typed columns; "
                f"got null typed columns: {null_columns}"
```

OK, for these two use cases, consider:

```python
ibis.Schema(
	{
		"s": "struct<n: null>",
		"a": "array<null>",
		"m1": "map<string: null>",
		"m2": "map<null: string>",
		"n": "null"
	}
).null_fields
```

What should this be? Could return something ibis-internal that isn't really machine-interpretable, and only for this error message, that is like [jq's DSL](https://jqlang.github.io/jq/manual/#array-object-value-iterator), like `("s.n", "a<items>",  "m1<values>", "m2<keys>", "n")`? I think that would suffice for our two needs.

Reply:
The amount of work here isn't justified IMO for nested types.

Can you show an example from any of our backends where it's possible to construct a nested null type let alone use them for anything?

I don't think this bit of code needs to change at all. The property is computing the fields of the schema that have a null type, that's it.